Add optimized batch update of hotspot indexes to release-6.0.0-dmr (#8671)
pingcap/docs: releases/release-6.0.0-dmr.md

    [User document](/functions-and-operators/expressions-pushed-down.md#add-to-the-blocklist), [#30738](https://github.com/pingcap/tidb/issues/30738)

> - Optimized batch update of hotspot indexe

Reviewer: ```suggestion
- Optimization of hotspot index
```

>     When writing data in batches to TiDB, the single index, auto-incremented by hotspot, causes a performance bottleneck easily, affecting the overall write throughput. In v6.0.0, TiDB supports sharding the index to different nodes through the `tidb_shard` function, which improves the write performance. This is an application-friendly solution and does not require modifying the original query conditions. You can use the solution under the scenarios of high write throughput, point queries, and batch point queries. Note that you need to use this feature after verifying it because using the data that is sharded by range query might cause performance rollback.

Reviewer: ```suggestion
    When writing data in batches monotonically to the secondary index makes it a hot spot index, causes a performance bottleneck easily, affecting the overall write throughput. In v6.0.0, TiDB supports scattering the hot spot index by the `tidb_shard` function, which improves the write performance. `tidb_shard` only affects the secondary unique index currently. This is an application-friendly solution and does not require modifying the original query conditions. You can use the solution under the scenarios of high write throughput, point queries, and batch point queries. Note that you need to use this feature after verifying it because using the data that is sharded by range query might cause performance rollback.
```

>     When writing monotonically increasing data in batches to the secondary index, it makes a hotspot index and causes a performance bottleneck easily, affecting the overall write throughput. In v6.0.0, TiDB supports sharding the index to different nodes through the `tidb_shard` function, which improves the write performance. Currently, `tidb_shard` only affects the secondary unique index. This is an application-friendly solution and does not require modifying the original query conditions. You can use the solution under the scenarios of high write throughput, point queries, and batch point queries. Note that you need to use this feature after verifying it because using the data that is sharded by range query might cause performance rollback.

Reviewer 2: ```suggestion
    When writing monotonically increasing data in batches to the secondary index, it makes a hotspot index and causes a performance bottleneck easily, affecting the overall write throughput. In v6.0.0, TiDB supports sharding the index to different nodes through the `tidb_shard` function, which improves the write performance. Currently, `tidb_shard` only affects the secondary unique index. This is an application-friendly solution and does not require modifying the original query conditions. You can use the solution under the scenarios of high write throughput, point queries, and batch point queries. Note that you need to use this feature after verifying it because using the data that is sharded by range query might cause performance fallback.
```

>     When writing monotonically increasing data in batches to the secondary index, it makes a hotspot index and causes a performance bottleneck easily, affecting the overall write throughput. In v6.0.0, TiDB supports sharding the index to different nodes through the `tidb_shard` function, which improves the write performance. Currently, `tidb_shard` only affects the secondary unique index. This is an application-friendly solution and does not require modifying the original query conditions. You can use the solution under the scenarios of high write throughput, point queries, and batch point queries. Note that you need to use this feature after verifying it because using the data that is sharded by range query might cause performance fallback.

Reviewer 3: https://github.com/pingcap/docs-cn/pull/8775 does not mention "打散至其它节点".
```suggestion
    Writing monotonically increasing data in batches to the secondary index causes an index hotspot and affects the overall write throughput. Since v6.0.0, TiDB supports scattering the index hotspot using the `tidb_shard` function to improve the write performance. Currently, `tidb_shard` only takes effect on the unique secondary index. This application-friendly solution does not require modifying the original query conditions. You can use the solution in the scenarios of high write throughput, point queries, and batch point queries. Note that using the data that has been scattered by range queries in the application might cause a performance regression. Therefore, do not use this function in such cases without verification.
```

>     [User document](/functions-and-operators/tidb-functions.md#tidbshard), [#31040](https://github.com/pingcap/tidb/issues/31040)

Author: ```suggestion
    [User document](/functions-and-operators/tidb-functions.md#tidb_shard), [#31040](https://github.com/pingcap/tidb/issues/31040)
```
